examples/sparse_blas/compile_time_dispatching/sparse_blas_spmv_usm_mklcpu.cpp

```diff
@@ -137,23 +140,49 @@ int run_sparse_matrix_vector_multiply_example(const sycl::device &cpu_dev) {
     std::cout << "\t\t\tnrows = " << nrows << std::endl;
     std::cout << "\t\t\talpha = " << alpha << ", beta = " << beta << std::endl;
 
-    // create and initialize handle for a Sparse Matrix in CSR format
-    oneapi::mkl::sparse::matrix_handle_t handle = nullptr;
-
-    oneapi::mkl::sparse::init_matrix_handle(cpu_selector, &handle);
-
-    auto ev_set = oneapi::mkl::sparse::set_csr_data(cpu_selector, handle, nrows, nrows, nnz,
-                                                    oneapi::mkl::index_base::zero, ia, ja, a);
-
-    auto ev_opt = oneapi::mkl::sparse::optimize_gemv(cpu_selector, transA, handle, { ev_set });
-
-    auto ev_gemv =
-        oneapi::mkl::sparse::gemv(cpu_selector, transA, alpha, handle, x, beta, y, { ev_opt });
-
-    auto ev_release =
-        oneapi::mkl::sparse::release_matrix_handle(cpu_selector, &handle, { ev_gemv });
-
-    ev_release.wait_and_throw();
+    // Create and initialize handle for a Sparse Matrix in CSR format
+    oneapi::mkl::sparse::matrix_handle_t A_handle = nullptr;
+    oneapi::mkl::sparse::init_csr_matrix(cpu_selector, &A_handle, nrows, nrows, nnz,
+                                         oneapi::mkl::index_base::zero, ia, ja, a);
+
+    // Create and initialize dense vector handles
+    oneapi::mkl::sparse::dense_vector_handle_t x_handle = nullptr;
+    oneapi::mkl::sparse::dense_vector_handle_t y_handle = nullptr;
+    oneapi::mkl::sparse::init_dense_vector(cpu_selector, &x_handle, sizevec, x);
+    oneapi::mkl::sparse::init_dense_vector(cpu_selector, &y_handle, sizevec, y);
+
+    // Create operation descriptor
+    oneapi::mkl::sparse::spmv_descr_t descr = nullptr;
+    oneapi::mkl::sparse::init_spmv_descr(cpu_selector, &descr);
+
+    // Allocate external workspace
+    std::size_t workspace_size = 0;
+    oneapi::mkl::sparse::spmv_buffer_size(cpu_selector, transA, &alpha, A_view, A_handle, x_handle,
+                                          &beta, y_handle, alg, descr, workspace_size);
+    void *workspace = sycl::malloc_device(workspace_size, cpu_queue);
+
+    // Optimize spmv
+    auto ev_opt =
+        oneapi::mkl::sparse::spmv_optimize(cpu_selector, transA, &alpha, A_view, A_handle, x_handle,
+                                           &beta, y_handle, alg, descr, workspace);
+
```

Comment:
is this benefiting from `const std::vector<sycl::event>& deps = {}`  here ? so no events in is an empty deps list ?  or does it not take events ?

Reply:
It can take events but none are needed here.

Reply:
@spencerpatty, [init_csr_matrix](https://uxlfoundation.github.io/oneAPI-spec/spec/elements/oneMKL/source/domains/spblas/data_types/init_csr_matrix.html#onemkl-sparse-init-csr-matrix) and [spmv_buffer_size](https://uxlfoundation.github.io/oneAPI-spec/spec/elements/oneMKL/source/domains/spblas/operations/spmv.html#onemkl-sparse-spmv) are both synchronous; they don't take in a `dependencies` vector, and they don't return a `sycl::event`. Consequently here, `spmv_optimize` is the first asynchronous event submission.

Reply:
Resolving as this is answered.